filters/include/pcl/filters/voxel_grid_covariance.h

```diff
@@ -425,7 +454,12 @@ namespace pcl
         k_leaves.reserve (k);
         for (const int &k_index : k_indices)
         {
-          k_leaves.push_back (&leaves_[voxel_centroids_leaf_indices_[k_index]]);
+          auto voxel = leaves_.find(voxel_centroids_leaf_indices_[k_index]);
+          if (voxel == leaves_.end()) {
+            continue;
```

Comment:
should you call `--k` before continuing?

Reply:
Maybe replace `return k;` with `return k_leaves.size()`?

Reply:
I fixed the code according to @kunaltyagi